### ISSUE TITLE
4.0: 'no-unused-variable is deprecated....' message when using "extends": "tslint:recommended"

### ISSUE THREAD

**Issue by locobert** [palantir/tslint#1748]
When using TSLint 4.0 with a rule set that `"extends": "tslint:recommended"`, the following deprecation message is always shown

    no-unused-variable is deprecated. Use the compiler options --noUnusedParameters and --noUnusedLocals instead.

Removing the rule from `node_modules/tslint/lib/configs/recommended.js` (where it is set to `false`) fixes it

**Comment by devoto13** [comment]
If we need to have this warning, can we at least show it ONLY, when rule is enabled. It is pretty inconvenient to copy whole config file to get rid of it.

**Comment by Ungolianth** [comment]
The notice should be suppressed, once the rule is actively disabled in the config:
```json
{
    "extends": "tslint:latest",
    "rules": {
        "no-unused-variable": false
    }
}
```

Right now, the notice still appears (and doesn't fair well with some JetBrains-IDEs).

**Comment by locobert** [comment]
@Ungolianth not for me: as long as the rule is mentioned (including as `"no-unused-variable": false,` in `tslint/lib/configs/recommended.js`, which is also used in `latest`), the warning appears

**Comment by Ungolianth** [comment]
@locobert yes, it was ment to be "It **should** work like this, but right now it does not".

Actively changing the `node_modules/tslint/lib/configs/recommended.js` cannot be the _ultimate_ solution, until "no-unused-variable" is removed officially from `tslint:recommended`, as your changes are overwritten with every (minor) update of tslint...
And the packages behavior will not be consistent for teams.

**Comment by locobert** [comment]
:) Yes, modifying `node_modules/tslint/lib/configs/recommended.js` was not meant as a solution or even workaround, it was meant to indicate the source of the problem

**Comment by adidahiya** [comment]
sorry about that, v4.0.1 should have the fix

**Comment by e-cloud** [comment]
It also prevent the tslint service in webstorm from working.
![image](https://cloud.githubusercontent.com/assets/5255354/20585015/c20016dc-b232-11e6-9c01-1f11bdd6352e.png)

**Comment by Ungolianth** [comment]
Did you update your tslint package?

All JetBrains IDEs show this behaviour (I use PHPStorm and IntelliJ), as they cannot map this message to a specific file. Removing the `no-unused-variable` rule from your project now (v4.0.1) also fixes your IDE.

**Comment by e-cloud** [comment]
In fact, i prefer to turn on this rule because it can't be disabled for some code with common tag, which the tsc can't achieve for now.

**Comment by adidahiya** [comment]
@e-cloud what do you mean by "common tag" here? With `tsc`, you can opt out of `noUnused...` checking by prefixing variable or parameter identifiers with underscore, e.g. `_unusedVar`

**Comment by e-cloud** [comment]
sorry, typo. I mean use tslint comment flag to disable something.

**Comment by nomaed** [comment]
The problem with using the flag in `tsc` is that it causes an error during build, while the `tslint` rules is more configurable and can be used to produce warnings only.

**Comment by LukeSkyw** [comment]
Problem still here with:
```
"tslint": "^4.4.2",
"tslint-microsoft-contrib": "^4.0.0",
```
